plugin/caffe/caffe_operator_util.cc

```diff
@@ -0,0 +1,158 @@
+#include <caffe/layers/base_conv_layer.hpp>
+#include <caffe/layers/base_data_layer.hpp>
+#include <caffe/layers/batch_norm_layer.hpp>
+#include <caffe/layers/batch_reindex_layer.hpp>
+#include <caffe/layers/bias_layer.hpp>
+#include <caffe/layers/bnll_layer.hpp>
+#include <caffe/layers/concat_layer.hpp>
+#include <caffe/layers/contrastive_loss_layer.hpp>
+#include <caffe/layers/conv_layer.hpp>
+#include <caffe/layers/crop_layer.hpp>
+#include <caffe/layers/data_layer.hpp>
+#include <caffe/layers/deconv_layer.hpp>
+#include <caffe/layers/dropout_layer.hpp>
+#include <caffe/layers/dummy_data_layer.hpp>
+#include <caffe/layers/eltwise_layer.hpp>
+#include <caffe/layers/elu_layer.hpp>
+#include <caffe/layers/embed_layer.hpp>
+#include <caffe/layers/euclidean_loss_layer.hpp>
+#include <caffe/layers/exp_layer.hpp>
+#include <caffe/layers/filter_layer.hpp>
+#include <caffe/layers/flatten_layer.hpp>
+#include <caffe/layers/hdf5_data_layer.hpp>
+#include <caffe/layers/hdf5_output_layer.hpp>
+#include <caffe/layers/hinge_loss_layer.hpp>
+#include <caffe/layers/im2col_layer.hpp>
+#include <caffe/layers/image_data_layer.hpp>
+#include <caffe/layers/infogain_loss_layer.hpp>
+#include <caffe/layers/inner_product_layer.hpp>
+#include <caffe/layers/input_layer.hpp>
+#include <caffe/layers/log_layer.hpp>
+#include <caffe/layers/loss_layer.hpp>
+#include <caffe/layers/lrn_layer.hpp>
+#include <caffe/layers/memory_data_layer.hpp>
+#include <caffe/layers/multinomial_logistic_loss_layer.hpp>
+#include <caffe/layers/mvn_layer.hpp>
+#include <caffe/layers/neuron_layer.hpp>
+#include <caffe/layers/parameter_layer.hpp>
+#include <caffe/layers/pooling_layer.hpp>
+#include <caffe/layers/power_layer.hpp>
+#include <caffe/layers/prelu_layer.hpp>
+#include <caffe/layers/reduction_layer.hpp>
+#include <caffe/layers/relu_layer.hpp>
+#include <caffe/layers/reshape_layer.hpp>
+#include <caffe/layers/scale_layer.hpp>
+#include <caffe/layers/sigmoid_cross_entropy_loss_layer.hpp>
+#include <caffe/layers/sigmoid_layer.hpp>
+#include <caffe/layers/silence_layer.hpp>
+#include <caffe/layers/slice_layer.hpp>
+#include <caffe/layers/softmax_layer.hpp>
+#include <caffe/layers/softmax_loss_layer.hpp>
+#include <caffe/layers/split_layer.hpp>
+#include <caffe/layers/spp_layer.hpp>
+#include <caffe/layers/tanh_layer.hpp>
+#include <caffe/layers/threshold_layer.hpp>
+#include <caffe/layers/tile_layer.hpp>
+#include <caffe/layers/window_data_layer.hpp>
+
+#if MSHADOW_USE_CUDNN == 1
+#include "caffe/layers/cudnn_conv_layer.hpp"
+#include "caffe/layers/cudnn_lcn_layer.hpp"
+#include "caffe/layers/cudnn_lrn_layer.hpp"
+#include "caffe/layers/cudnn_pooling_layer.hpp"
+#include "caffe/layers/cudnn_relu_layer.hpp"
+#include "caffe/layers/cudnn_sigmoid_layer.hpp"
+#include "caffe/layers/cudnn_softmax_layer.hpp"
+#include "caffe/layers/cudnn_tanh_layer.hpp"
+#endif
+
+namespace mxnet {
+namespace op {
+
+
+MXNET_REGISTER_CAFFE_LAYER(InnerProduct, caffe::InnerProductLayer);
```

Comment:
The purpose of caffe plugin is to let users migrate from caffe easily. That means they'll be using this to run their own layers, which is not defined here.

It's more difficult for them to add code here and recompile than, say, specify that in a parameter, like is done for torch module.
We can even add a logic in caffe_converter that counts the number of input a layer is having and feed it to the symbol.

tl;dr; I think adding users specify how many input this layer has is better than hard code it here.